soccer/motion/MotionControl.cpp

```diff
@@ -261,8 +259,9 @@ void MotionControl::_targetBodyVel(Point targetVel) {
     }
 
     // set control values
-    _robot->control->set_xvelocity(targetVel.x() * _x_multiplier->value());
-    _robot->control->set_yvelocity(targetVel.y());
+    _context->robotIntents[_robot->shell()].setpoints.xvelocity =
+        targetVel.x() * _x_multiplier->value();
+    _context->robotIntents[_robot->shell()].setpoints.yvelocity = targetVel.y();
```

Comment:
This is my only concern with this PR. I don't want both gameplay and motion control to be publishing _partial_ structs, because a large part of the point of the refactor is allowing us to transition smoothly to a pub/sub where you're publishing an entire message at a time (each message needs to be generated _in whole_ before it is "published")

Maybe have a separate struct for `RobotMotionSetpoints` that's published by motion control, instead of writing back into the intent? I'm just spitballing, this is one of the annoying details of this change that I don't think I had thought through entirely, so I'd love to hear ideas.